Implementación de la incidencia JOKO-56

MathiMartinez00 opened the pull request:
Se implementó la incidencia [JOKO-56](https://sodepsa.atlassian.net/browse/JOKO-56?atlOrigin=eyJpIjoiM2MxZDA1NjVhNTk2NDBjMDg5YWE5MjM3MWEyMWFmZmUiLCJwIjoiaiJ9).

Resumen de los cambios realizados:

- El sidebar ahora es complemetamente responsive, para ello se creo el componente CollapsedSidebar el cual sirve de wrapper. El botón para hacer aparecer el sidebar solo se muestra en breakpoints inferiores a md.
- Se adaptó el diseño del layout para que sea similar al sugerido por Rodrigo.
- Se agregó el componente SidebarItem para modularizar el Sidebar y evitar repetir clases.
- Se creo el tipo Module para contener dentro de este información de los módulos.
- Se agregó la documentación de todos los componentes en layout.tsx a README.md.

MathiMartinez00 committed:
Empezar layout principal con Tailwind

MathiMartinez00 committed:
Agregar navbar vertical

Actualmente el navbar solo aparece desde el breakpoint max-sm.
TODO: Agregar un boton para hacerlo colapsar al clickear.

MathiMartinez00 committed:
Hacer que el layout se expanda toda la altura

MathiMartinez00 committed:
Arreglar columnas en max-sm y cambiar color de fondo

MathiMartinez00 committed:
Agregar TODO y cambios de cursor

MathiMartinez00 committed:
Agregar algo al TODO

MathiMartinez00 committed:
Agregar react-feather a las dependencias

Se opto por esta libreria segun lo acordado con Rodrigo

MathiMartinez00 committed:
Agregar iconos al layout

Se agregaron los iconos User y Upload al layout siguiendo el ejemplo proveido en la tarea de Jira.

MathiMartinez00 committed:
Agregar un icono al layout

Se agrego AlignJustify para el icono de colapsar la barra de navegacion

MathiMartinez00 committed:
Cambiar de anchor a next/link

MathiMartinez00 committed:
Hacer el layout sea un layout de verdad

MathiMartinez00 committed:
Agregar tipos para TypeScript

MathiMartinez00 committed:
Hacer que los <li> en index.tsx sean clickeables

MathiMartinez00 committed:
Eliminar archivo no usado

MathiMartinez00 committed:
Agregar paginas de prueba para el layout

MathiMartinez00 committed:
Agregar ruta a lista de rutas a ignorar para el layout

Se agrego /signup a la lista de rutas ignoradas ya que forma parte del
formulario utilizado para iniciar sesion.
Tambien se agrego documentacion sobre la funcion creada para esto.

MathiMartinez00 committed:
Cambiar nombre a funcion

MathiMartinez00 committed:
Dividir componentes del Layout

Tambien se agrego documentacion al respecto en README.md y al inicio de cada componente.

AlainVega committed:
Merge branch 'develop' of https://github.com/MathiMartinez00/security-starter-react into develop

AlainVega committed:
Add login (que se borro sin querer) y un link entre index y login

MathiMartinez00 committed:
Empezar cambios

Se agregaron colores necesarios en tailwind.config.js
Se agrego un logo tentativo a la aplicacion
Se cambio el color de fondo para todo el layout

MathiMartinez00 committed:
Agregar secciones al sidebar

MathiMartinez00 committed:
Mover el link a login

Ahora esta en la cabecera del layout

MathiMartinez00 committed:
Agregar estado a modulos en el sidebar

MathiMartinez00 committed:
Agregar SidebarItem

SidebarItem representa una fila en el Sidebar.
Es basicamente un wrapper de los contenidos pero hecho componente para que sea reutilizable.

MathiMartinez00 committed:
Agregar tipo Module

Este tipo de utilizado para pasar informacion del modulo seleccionado a diferentes componentes del layout, tiene los siguientes componentes:
id: Numero arbitrario para identificar al modulo.
name: Nombre del modulo para mostrarlo.

MathiMartinez00 committed:
Agregar titulo a la cabecera

Este titulo es el nombre del modulo y esta en el componente Header de components/layout.tsx.

MathiMartinez00 committed:
Sacar cosas del TODO

MathiMartinez00 committed:
Iniciar cabecera responsive

MathiMartinez00 committed:
Ocultar parte de la cabecera en breakpoints inferiores a md

MathiMartinez00 committed:
Empezar sidebar desplegable

MathiMartinez00 committed:
Agregar CollapsedSidebar

MathiMartinez00 committed:
Bugfix condicion de display de sidebar

MathiMartinez00 committed:
El boton AlignJustify ahora funciona

MathiMartinez00 committed:
Cambiar estado por defecto de sidebar

MathiMartinez00 committed:
Bugfix CollapsedSidebar

El sidebar ahora colapsa correctamente.
Para ello, se agrego el prop classes al componente Sidebar para dar clases condicionalmente dependiendo de si el sidebar esta colapsado o no.

MathiMartinez00 committed:
Actualizar documentacion

La documentacion de los componentes en layout.tsx fue actualizada para reflejar los cambios hechos.
Ademas fueron movidos para que esten por encima de la definicion de los componentes.

MathiMartinez00 committed:
Actualizar README.md

Fue actualizado con la informacion de todos los componentes añadidos

MathiMartinez00 committed:
Corregir errores en README.md

MathiMartinez00 committed:
Merge branch 'main' into JOKO-56-modelo-figma

MathiMartinez00 committed:
Eliminar link innecesario a login

rodrigojv commented on components/layout.tsx:
```diff
@@ -1,79 +1,236 @@
+  isSidebarCollapsed: boolean,
+  setSidebarCollapsed: Function,
+}) {
+  if (isSidebarCollapsed == true) {
+    return(
+      <></>
+    )
+  }
```
Se puede retornar null simplemente.

MathiMartinez00 replied:
Cambio hecho en el último commit.

rodrigojv commented on components/layout.tsx:
```diff
@@ -1,79 +1,236 @@
+  children: React.ReactNode,
+}) {
+  return (
+    <li className={(selectedModule.id === moduleData.id ?
```
Una mejora que sugiero para esto es usar la ruta actual para saber que estilo aplicarle de color de fondo. Esto puede tratar de implementarse en otros PR. https://stackoverflow.com/questions/58022046/get-url-pathname-in-nextjs

rodrigojv commented on components/layout.tsx:
```diff
@@ -1,79 +1,236 @@
+                name: 'Settings',
+              }}
+            >              
+              <Settings className="inline-block" />
```
El texto puede formar parte del children de este componente en lugar de usar moduleData.name.

```jsx
<SidebarItem>
<Settings> <div className="px-4 text-lg">Settings</div>
</SidebarItem>
```

MathiMartinez00 replied:
La razón por la que moduleData tiene el nombre del módulo es debido a que el estado selectedModule (que está definido en PageLayout) es compartido con el componente Header para poder mostrar el nombre del módulo como título. 

Se realizó el cambio propuesto pero si se quiere eliminar esa información del módulo entonces se debe cambiar como se muestra el título por página y no sé si debo hacerlo. Aguardó confirmación.

rodrigojv commented on components/layout.tsx:
```diff
@@ -1,79 +1,236 @@
+  isSidebarCollapsed: boolean,
+  setSidebarCollapsed: Function,
+}) {
+  if (isSidebarCollapsed == true) {
```
En JS se recomienda siempre utilizar el triple `===`

MathiMartinez00 replied:
Cambio hecho en el último commit.

rodrigojv commented on components/layout.tsx:
```diff
@@ -1,79 +1,236 @@
+
+/**
+ * Sidebar del layout.  
+ * Funciona como una lista <ul></ul> con la clase overflow-y-auto donde cada elemento <li></li> representa un item dentro
+ * del sidebar.  
+ * Ademas, cada elemento que no sea el titulo del sidebar tiene un next/link alrededor del <li></li> para
+ * poder redireccionar a esa pagina.  
+ * Recibe:  
+ * selectedModule: Objeto tipo Module con informacion del actualmente seleccionado.  
+ * setSelectedModule: Funcion para cambiar selectedModule.  
```
Está genial la documentación de los componentes, pero algo más práctico sería definir los props como tipos de datos de Typescript y eso también contribuye a tener una documentación, al mismo tiempo que hacer el código más robusto.

MathiMartinez00 replied:
Lo tendré en cuenta, gracias.

MathiMartinez00 committed:
Aplicar cambios sugeridos

Las comparaciones ahora utilizan el operando ===
El retorno de CollapsedSidebar ahora es null cuando el sidebar esta colapsado

MathiMartinez00 committed:
Agregar el nombre del modulo al cuerpo de SidebarItem

rodrigojv commented:
@MathiMartinez00 aprobados los cambios. Resolvamos los conflictos para poder mezclar.

MathiMartinez00 committed:
Merge branch 'main' into JOKO-56-modelo-figma

MathiMartinez00 commented:
Conflictos resueltos.
